### ISSUE TITLE
rich_click.confirm does not translate color tags

### ISSUE THREAD

**Issue by pkulev** [ewels/rich-click#214]
<img width="617" alt="изображение" src="https://github.com/user-attachments/assets/42062430-5f36-4147-aae6-dae07ee8f625">

```python
In [8]: rich_click.__version__
Out[8]: '1.8.3'
```

**Comment by dwreeves** [comment]
This is related to #73 and #29.

I would say this is not a bug, it is intentional. We do not rich mark-up `click.confirm()` (and `click.prompt()`) for the same reason we do not rich mark-up `click.echo()`. The rationale is described in this comment here: https://github.com/ewels/rich-click/pull/29#issuecomment-2081474377

Rich has an internal feature for this as well: https://rich.readthedocs.io/en/stable/prompt.html

```python
import rich.prompt
yes = rich.prompt.Confirm.ask("Confirm?")
```

I'll keep this issue open for now, but I am leaning toward this not being something we will do. IMO, users interested in a `click.confirm()` or `click.prompt()` that is marked up should use the `rich.prompt` module.

That said, we still don't document this intentional decision, and we should point users toward this alternative.

**Comment by dwreeves** [comment]
We are clarifying in the docs that this is intentional: #216.

That said, upon doing some research, I ran into `@click.option(..., prompt=True)` and `confirm=True` as well.

I am thinking we may want to make a `RichOption` that uses the `RichHelpConfiguration()` to determine whether to use rich for prompts/confirms. Although, I am not sure if this should be a 1.9 or 2.0 roadmap item. Possibly the solution would be: by default for 1.9 we have it set to false, for 2.0 we set to true. Not sure.
